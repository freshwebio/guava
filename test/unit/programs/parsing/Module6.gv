module Order exposing { total }

function total() :: Float {
  540.34
}