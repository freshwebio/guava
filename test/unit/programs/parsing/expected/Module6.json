{
  "description": "should parse a program with a function that returns a floating point constant with a single value inline type definition",
  "module": {
    "name": "Order",
    "exposes": {
      "exposed": ["total"]
    },
    "moduleStatements": {
      "statementList": [{
        "kind": "function",
        "fncStatement": {
          "kind": "functionWithInlineTypeDef",
          "functionName": "total",
          "defArgs": {
            "args": []
          },
          "functionTypeDefValueTypes": [{
            "kind": "valueTypeSingle",
            "valueTypeName": "Float"
          }],
          "functionExpr": {
            "kind": "arithmeticExpr",
            "arithmeticExpr": {
              "kind": "floatConst",
              "floatConstValue": 540.34
            }
          }
        }
      }]
    }
  }
}